tools/resourcedocsgen/cmd/metadata.go

```diff
@@ -276,11 +276,18 @@ func packageMetadataFromGitHubCmd(metadataDir, packageDocsDir *string) *cobra.Co
 			// Normalize end of line representation
 			content = bytes.ReplaceAll(content, []byte("\r\n"), []byte("\n"))
 
-			if rest, ok := bytes.CutPrefix(bytes.TrimLeft(content, "\n\t\r "), []byte("---\n")); ok {
-				content = append([]byte(`---
-# WARNING: this file was fetched from `+url+`
+			frontmatter := []byte(`---
+# WARNING: this file was fetched from ` + url + `
 # Do not edit by hand unless you're certain you know what you are doing!
-`), rest...)
+`)
+			editURL := computeEditURLFromGitHubUserContentURL(url)
+			if editURL == nil {
+				return fmt.Errorf("expected URL %s to be a valid GitHub user content URL", url)
```

Comment:
Nit, this should be `%q` (print as a Go string) so it includes the `""`. This is especially helpful when `url` is empty.
```suggestion
				return fmt.Errorf("expected URL %q to be a valid GitHub user content URL", url)
```